runtime/runtime-core/common/src/aws/smithy/kotlin/runtime/util/Convenience.kt

```diff
@@ -35,3 +35,25 @@ public fun truthiness(value: Any?): Boolean = when (value) {
     null -> false
     else -> true
 }
+
+@InternalApi
+public fun List<Number>.max(): Double? =
+    @Suppress("UNCHECKED_CAST")
+    (this as List<Double>).maxOrNull()
```

Comment:
**Question**: Why is casting to `Double` correct here? Shouldn't the max of `[1L, 2L, 3L]` be `3L` instead of `3.0`?